docs/websockets.md

```diff
@@ -95,6 +95,55 @@ effect type      class name
 ================ ==========================================
 ```
 
+## WebSockets as Ox Source and Sink
+
+[Ox](https://ox.softwaremill.com) is a Scala 3 toolkit that allows you to handle concurrency and resiliency in direct-style, leveraging Java 21 virtual threads.
+If you're using Ox with `sttp`, you can use the `DefaultSyncBackend` from `sttp-core` for HTTP communication. An additional `ox` module allows handling WebSockets 
+as Ox `Source` and `Sink`:
+
+```
+// sbt dependency
+"com.softwaremill.sttp.client4" %% "ox" % "@VERSION@",
+```
+
+```scala 
+import ox.*
+import ox.channels.{Sink, Source}
+import sttp.client4.*
+import sttp.client4.impl.ox.ws.* // import to access asSourceAnkSink
+import sttp.client4.ws.SyncWebSocket
+import sttp.client4.ws.sync.*
+import sttp.ws.WebSocketFrame
+
+def useWebSocket(ws: SyncWebSocket): Unit =
+    supervised {
+      val (wsSource, wsSink) = asSourceAndSink(ws) // (Source[WebSocketFrame], Sink[WebSocketFrame])
+      // ...
+    }
+
+val backend = DefaultSyncBackend()
+basicRequest
+  .get(uri"wss://ws.postman-echo.com/raw")
+  .response(asWebSocket(useWebSocket))
+  .send(backend)
+```
+
+See the [full example here](https://github.com/softwaremill/sttp/blob/master/examples/src/main/scala/sttp/client4/examples3/WebSocketOx.scala).
+
+Make sure that the `Source` is contiunually read. This will guarantee that server-side Close signal is received and handled.
+If you don't want to process frames from the server, you can at least handle it with a `fork { source.drain() }`.
+
+You don't need to manually call `ws.close()` when using this approach, this will be
+handled automatically underneath, according to following rules:
+ - If the request `Sink` fails with an error, the `Source` is automatically completed, sending a `Close` frame to
```

Comment:
isn't this the other way round? what's the "request Sink" here?

Reply:
I think what I don't get here is why, when the sink fails, source is completed (that's ok - meaning it's `done()`?), and this somehow sends the `Close` frame? Logically, you send frames through the requests sink, no? :)

Reply:
Yeah, this doesn't sound accurate. The logic is that, if the sink channel is closed with an error (for example, something unhandled failed in the user's pipeline), a Close frame is sent, and the request processing loop ends. I shall rephrase this.
> source is completed (that's ok - meaning it's done()?)
Yes, it will be `done()`, so that the responses loop stops receiving.

Reply:
ok, thanks :)